### DISCUSSION TITLE
How to select 2 of the outputs out of 3 outputs of a multi-task GP for qEHVI?

### DISCUSSION THREAD

**Discussion by sheikhahnaf** [meta-pytorch/botorch#2415] [Q&A] [answered]
Hi,
I am constructing a multi-task GP with 3 tasks but I want to maximize only task 1 and 2 not 3 through qEHVi? could you please point me to the right direction, for using which argument of qEHVI to make it do maximixzation only  task 1 & 2? Is it through fixedfeatureacquisitionfunction? or is fixed featureacacquisitionfunction for selecting fixed input dimensions for use in optimization? could anyoune please clarify my confusion?

**Comment by kstone40** [comment]
I believe you can use GenericMultiOutputObjective to convert a callable which simply indexes the first 2 outputs.

def select(sample):
 sample_out = sample[..., :2]
 return sample_out

objective = GenericMultiOutputObjective(select)

Then pass objective to your aq function

**Comment by saitcakmak** [comment, marked as answer]
Hi @sheikhahnaf. @kstone40's suggestion above should work. Here's another alternative that is built into the MultiTaskGP model. If you specify `output_tasks = [1, 2]` while constructing `MultiTaskGP`, it will produce the posterior only for tasks 1 and 2. From there on, you can treat the model just like any other two-output GP in the acquisition functions.

**Reply by sheikhahnaf** [reply]
Thank you so much @kstone40  @saitcakmak  I just had another issue opened in this repo, where I was asking why my Multi-task GP is not working with qEHVI,I hope this will solve my problem of using MultiTaskGP object with qEHVI. I will try this
